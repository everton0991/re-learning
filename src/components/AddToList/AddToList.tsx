import React, { useState } from 'react'
import { Segment, Form, Button } from 'semantic-ui-react'
import { IState as Props } from 'pages/Home'

interface IProps {
  people: Props['people']
  setPeople: React.Dispatch<React.SetStateAction<Props['people']>>
}

function AddToList({ people, setPeople }: IProps) {
  const [input, setInput] = useState({
    name: '',
    age: '',
    note: '',
    image: ''
  })

  const handleChange = (
    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ): void => {
    setInput({
      ...input,
      [e.target.name]: e.target.value
    })
  }

  const handleClick = () => {
    if (!input.name || !input.age || !input.image) {
      return
    }

    setPeople([
      ...people,
      {
        name: input.name,
        age: parseInt(input.age),
        url: input.image,
        note: input.note
      }
    ])

    setInput({
      name: '',
      age: '',
      note: '',
      image: ''
    })
  }

  return (
    <Segment inverted color="violet">
      <Form inverted>
        <Form.Group widths="equal">
          <Form.Input
            fluid
            label="Name"
            type="text"
            placeholder="Name"
            value={input.name}
            onChange={handleChange}
            name="name"
          />

          <Form.Input
            fluid
            label="Age"
            type="number"
            placeholder="Age"
            value={input.age}
            onChange={handleChange}
            name="age"
          />

          <Form.Input
            fluid
            label="Image URL"
            type="text"
            placeholder="Image URL"
            value={input.image}
            onChange={handleChange}
            name="image"
          />
        </Form.Group>

        <Form.TextArea
          label="Note"
          placeholder="Note"
          value={input.note}
          onChange={handleChange}
          name="note"
        />

        <div className="flex justify-end">
          <Button color="yellow" onClick={handleClick}>
            Add to List
          </Button>
        </div>
      </Form>
    </Segment>
  )
}

export default AddToList
