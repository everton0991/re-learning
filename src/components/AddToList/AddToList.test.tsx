import { render } from '@testing-library/react'
import AddToList from './AddToList'

it('renders correctly', () => {
  const tree = render(
    <AddToList
      people={[{ name: '', age: 20, note: '', url: '' }]}
      setPeople={jest.fn()}
    />
  )
  expect(tree).toMatchSnapshot()
})
