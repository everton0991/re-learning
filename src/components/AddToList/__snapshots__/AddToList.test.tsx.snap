// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="AddToList"
      >
        <input
          class="AddToList-input"
          name="name"
          placeholder="Name"
          type="text"
          value=""
        />
        <input
          class="AddToList-input"
          name="age"
          placeholder="Age"
          type="number"
          value=""
        />
        <input
          class="AddToList-input"
          name="image"
          placeholder="Image URL"
          type="text"
          value=""
        />
        <textarea
          class="AddToList-input"
          name="note"
          placeholder="Note"
        />
        <button
          class="AddToList-btn"
        >
          Add to List
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="AddToList"
    >
      <input
        class="AddToList-input"
        name="name"
        placeholder="Name"
        type="text"
        value=""
      />
      <input
        class="AddToList-input"
        name="age"
        placeholder="Age"
        type="number"
        value=""
      />
      <input
        class="AddToList-input"
        name="image"
        placeholder="Image URL"
        type="text"
        value=""
      />
      <textarea
        class="AddToList-input"
        name="note"
        placeholder="Note"
      />
      <button
        class="AddToList-btn"
      >
        Add to List
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
